src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -28,76 +27,121 @@ public XmlFlowAnnotationSource (LinkContext context)
+					_context.LogMessage (MessageContainer.CreateWarningMessage ($"Attribute element doest not contain attribute 'assembly' in xml element '{attributeFullName}'", 2030));
+					continue;
+				} else if (assemblyName == "*")
+					attributeType = GetTypeFromName (concatAttribute);
+				else {
+					try {
+						assembly = GetAssembly (_context, AssemblyNameReference.Parse (assemblyName));
+					} catch (Exception) {
+						_context.LogMessage (MessageContainer.CreateWarningMessage ($"Could not resolve assembly '{assemblyName}' in attribute '{attributeFullName}' specified in the '{_xmlDocumentLocation}'", 2031));
+						continue;
 					}
+					attributeType = assembly.FindType (concatAttribute);
+				}
+				if (attributeType == null) {
+					_context.LogMessage (MessageContainer.CreateWarningMessage ($"Attribute type '{attributeFullName}' could not be found", 2032));
+					continue;
 				}
-			}
 
-			return parameterAnnotation;
-		}
+				ArrayBuilder<string> arguments = GetAttributeChildren (iterator.Current.SelectChildren ("argument", string.Empty));
+				MethodDefinition constructor = attributeType.GetConstructors ().FirstOrDefault (c => c.Parameters.Count == arguments.Count);
+				if (constructor == null) {
+					_context.LogMessage (MessageContainer.CreateWarningMessage ($"Could not find a constructor for type '{attributeType}' that receives '{arguments.Count}' arguments as parameter", 2022));
+					continue;
+				}
+				string[] xmlArguments = arguments.ToArray ();
+				bool recognizedArgument = true;
 
-		public DynamicallyAccessedMemberTypes GetPropertyAnnotation (PropertyDefinition property)
-		{
-			return _properties.TryGetValue (property, out var ann) ? ann : DynamicallyAccessedMemberTypes.None;
-		}
+				CustomAttribute attribute = new CustomAttribute (constructor);
+				for (int i = 0; i < xmlArguments.Length; i++) {
+					object argumentValue = null;
 
-		public DynamicallyAccessedMemberTypes GetReturnParameterAnnotation (MethodDefinition method)
-		{
-			return _methods.TryGetValue (method, out var ann) ? ann.ReturnAnnotation : DynamicallyAccessedMemberTypes.None;
+					if (constructor.Parameters[i].ParameterType.Resolve ().IsEnum) {
+						foreach (var field in constructor.Parameters[i].ParameterType.Resolve ().Fields) {
+							if (field.IsStatic && field.Name == xmlArguments[i]) {
+								argumentValue = Convert.ToInt32 (field.Constant);
+								break;
+							}
+						}
+						if (argumentValue == null) {
+							_context.LogMessage (MessageContainer.CreateWarningMessage ($"Could not parse argument '{xmlArguments[i]}' specified in '{_xmlDocumentLocation}' as a {constructor.Parameters[i].ParameterType.FullName}", 2021));
+							recognizedArgument = false;
+						}
+					} else {
+						switch (constructor.Parameters[i].ParameterType.MetadataType) {
+						case MetadataType.String:
+							argumentValue = xmlArguments[i];
+							break;
+						case MetadataType.Int32:
+							argumentValue = int.Parse (xmlArguments[i]);
```

Comment:
This will throw if the value is not an integer. Probably leading to linker failing.
We should call `TryParse` instead and log a warning if it fails.